uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/components/my-leave-calendar.js

```diff
@@ -68,6 +69,8 @@ define([
      * @param  {Array} leaveRequests - leave requests array from API
      */
     vm._indexLeaveRequests = function (leaveRequests) {
+      vm.leaveRequests = {};
```

Comment:
It is necessary because when the calendar reloads because of a leave request is created/edited vm.leaveRequests should clear before reassigning values, otherwise duplicate values are shown in the UI.